ui/src/components/visualizations/MapLibre/index.tsx

```diff
@@ -77,24 +56,39 @@ const MapLibre: React.FC<Props> = ({ className, fileContent, fileType }) => {
         mapStyle="https://basemaps.cartocdn.com/gl/positron-gl-style/style.json"
         style={{ width: "100%", height: "100%" }}
         maplibreLogo={true}
-        interactiveLayerIds={["polygon-layer", "line-layer"]}
+        interactiveLayerIds={["point-layer", "line-layer", "polygon-layer"]}
         onClick={(e) => {
-          setSelectedFeature(e.features?.[0]);
+          if (e.features?.[0]) {
+            onSelectedFeature(e.features[0]);
+          } else {
+            onSelectedFeature(undefined);
+            setShowFeaturePanel(false);
+          }
+        }}
+        onDblClick={(e) => {
+          if (e.features?.[0]) {
+            onSelectedFeature(e.features[0]);
+            onFlyToSelectedFeature?.(e.features[0]);
+            setShowFeaturePanel(true);
+          }
```

Comment:
Can you move both onClick and onDblClick code into callbacks? (`handleMapClick`, `handleMapDoubleClick`)